src/occwl/face.py

```diff
@@ -402,8 +422,13 @@ def point_to_parameter(self, pt):
         Returns:
             np.ndarray: UV-coordinate
         """
-        uv = ShapeAnalysis_Surface(self.surface()).ValueOfUV(
-            gp_Pnt(pt[0], pt[1], pt[2]), 1e-9
+        loc = TopLoc_Location()
+        surf = BRep_Tool_Surface(self.topods_shape(), loc)
+        gp_pt = gp_Pnt(pt[0], pt[1], pt[2])
+        inv = loc.Transformation().Inverted()
```

Comment:
Would inverting the transform in every call make things slower? This is the right thing to do though. We could support sampling multiple points at once in future.

Reply:
The unit test was catching the need to do this.

Probably the transform knows when it is ortho-normal and can use the transpose/inverse trick.

Open Cascade is clearly designed for a workflow where you handle the transforms at a higher level.  It's a total mess.   When trying to write wrapper libraries over the top then providing an API which does a batch of points is probably the best we can hope for.